storm-client/src/jvm/org/apache/storm/dependency/DependencyUploader.java

```diff
@@ -154,11 +154,22 @@ private boolean uploadDependencyToBlobStore(String key, File dependency)
             acls.add(new AccessControl(AccessControlType.OTHER,
                                        BlobStoreAclHandler.READ));
 
-            AtomicOutputStream blob = getBlobStore().createBlob(key, new SettableBlobMeta(acls));
-            Files.copy(dependency.toPath(), blob);
-            blob.close();
+            AtomicOutputStream blob = null;
+            try {
+                blob = getBlobStore().createBlob(key, new SettableBlobMeta(acls));
+                Files.copy(dependency.toPath(), blob);
+                blob.close();
 
-            uploadNew = true;
+                uploadNew = true;
```

Comment:
shouldn't we call blob = null here to prevent a dupe cancel after close?

Reply:
Actually we probably want to remove the call to `close`, since it will happen in the `finally` block now.